grizzly/replay/replay.py

```diff
@@ -156,121 +158,139 @@ def _dyn_close():  # pragma: no cover
             server_map.set_dynamic_response("grz_harness", lambda: self._harness, mime_type="text/html")
 
         success = False
-        test_count = len(self.testcases)
-        for _ in range(repeat):
-            self.status.iteration += 1
-            if self.target.closed:
-                LOG.info("Launching target...")
-                if self._harness is None:
-                    location = self._runner.location(
-                        "/grz_current_test",
-                        self.server.port)
-                else:
-                    location = self._runner.location(
-                        "/grz_harness",
-                        self.server.port,
-                        close_after=self.target.rl_reset * test_count,
-                        forced_close=self.target.forced_close)
-                try:
-                    # The environment from the initial testcase is used because
-                    # a sequence of testcases is expected to be run without
-                    # relaunching the Target to match the functionality of
-                    # Grizzly. If this is not the case each TestCase should
-                    # be run individually.
-                    self._runner.launch(location, env_mod=self.testcases[0].env_vars)
-                except TargetLaunchError:
-                    LOG.error("Target launch error. Check browser logs for details.")
+        unpacked = list()
+        try:
+            LOG.debug("unpacking testcases (%d)...", test_count)
+            for test in testcases:
+                dst_path = mkdtemp(prefix="tc_", dir=grz_tmp("serve"))
+                test.dump(dst_path)
+                unpacked.append(dst_path)
```

Comment:
Can these two lines be swapped? If there's a problem dumping a testcase, the tempdir for that testcase wouldn't be in `unpacked` yet, and would be left behind.

Reply:
Good call, thanks.